lib/src/modules/compiler/template_parser/GRAMMAR.md

```diff
@@ -0,0 +1,38 @@
+#Template Grammar
+
+###Todos
+* Can you interpolate comments?
+* Can you have multiple pipes?
+* What is the best way to handle the additional constraints on expressions?
+* What are we forgetting?
+
+
+```dart
+Node           = VoidTag | OpenTag {Node} CloseTag | Comment | Text;
+OpenTag        = "<" TagName WhiteSpace {WhiteSpace} {Attribute WhiteSpace {WhiteSpace}} ">";
+VoidTag        = "<" TagName WhiteSpace {WhiteSpace} {Attribute WhiteSpace {WhiteSpace}} "/>";
```

Comment:
Just a nit/question/clarification: Are we saying all void tags _must_ match `<tag />'? What about implicit voids, i.e. `<br>`?